minifi/minifi-toolkit/minifi-toolkit-configuration/src/main/java/org/apache/nifi/minifi/toolkit/configuration/json/TransformNifiCommandFactory.java

```diff
@@ -101,4 +104,17 @@ private void persistFlowJson(VersionedDataflow flow, String flowJsonPath) throws
                     ConfigMain.ERR_UNABLE_TO_SAVE_CONFIG, e);
         }
     }
+
+    private void setDefaultValues(VersionedDataflow versionedDataflow) {
+        versionedDataflow.setMaxTimerDrivenThreadCount(CorePropertiesSchema.DEFAULT_MAX_CONCURRENT_THREADS);
+        versionedDataflow.getRootGroup().getRemoteProcessGroups().forEach(this::setDefaultValues);
```

Comment:
I think we should traverse all child process groups recursively, so all the necessary components' properties will be set. A possible implementation could be:
```
    private void setDefaultValues(VersionedDataflow versionedDataflow) {
        versionedDataflow.setMaxTimerDrivenThreadCount(CorePropertiesSchema.DEFAULT_MAX_CONCURRENT_THREADS);
        setDefaultValues(versionedDataflow.getRootGroup());
    }

    private void setDefaultValues(VersionedProcessGroup versionedProcessGroup) {
        versionedProcessGroup.getRemoteProcessGroups().forEach(this::setDefaultValues);
        versionedProcessGroup.getProcessGroups().forEach(this::setDefaultValues);
    }
```

Reply:
Thanks for your suggestion @briansolo1985, I'm happy to apply it.